SpoonTV/Scenes/SearchScene/SearchViewController.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  SearchViewController.swift
+//  SpoonTV
+//
+//  Created by Nam Ngây on 9/15/20.
+//  Copyright © 2020 Nam Ngây. All rights reserved.
+//
+
+import UIKit
+import RxCocoa
+import RxSwift
+import Reusable
+import MGArchitecture
+import MGLoadMore
+
+final class SearchViewController: UIViewController, BindableType {
+    
+    @IBOutlet private weak var searchBar: UISearchBar!
+    @IBOutlet private weak var tableView: LoadMoreTableView!
+    @IBOutlet private weak var segmentedControl: UISegmentedControl!
+    
+    var viewModel: SearchViewModel!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configView()
+     }
+    
+    func configView() {
+        tableView.do {
+            $0.register(cellType: SearchTableViewCell.self)
+            $0.rowHeight = 125
+        }
+    }
+    
+    func bindViewModel() {
+        let input = ViewModelType.Input(textSearch: searchBar.rx.text.orEmpty.asDriver(),
+                                        loadTrigger: Driver.just(()),
+                                        reloadTrigger: tableView.refreshTrigger,
+                                        loadMoreTrigger: tableView.loadMoreTrigger,
+                                        filterTrigger: segmentedControl.rx.selectedSegmentIndex.asDriver(),
+                                        selectTrigger: tableView.rx.itemSelected.asDriver())
+        
+        let output = viewModel.transform(input)
+        
+        switch segmentedControl.selectedSegmentIndex {
```

Comment:
nếu như theo logic hiện tại của em thì ở segment nào sẽ sử dụng output đó vậy thì nên add filter vào hơn cho nó dễ nhìn kiểu như này

output.searchMovie
.filter { [unowned self] in 
self.segmentedControl.selectedSegmentIndex == 0
}
                .drive(tableView.rx.items) { tableView, index, item in
                    let indexPath = IndexPath(item: index, section: 0)
                    let cell = tableView.dequeueReusableCell(for: indexPath, cellType: SearchTableViewCell.self)
                    cell.setContentMovie(item)
                    return cell
                }
            .disposed(by: rx.disposeBag)


tương tự cái bên dưới